### PR TITLE
The application shows a message when the gps isn't enabled

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/20" target="_blank">ReviewNinja

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -130,4 +130,12 @@
     <!--user location-->
     <string name="viewer_my_location_name">My_Location</string>
     <string name="viewer_self_icon_assets_path">mipmap/target-center-red.png</string>
+
+    <!-- GPS messages -->
+    <string name="gps_off_title">GPS is disabled</string>
+    <string name="gps_off_message">Your GPS seems to be disabled, do you want to enable it?</string>
+
+    <!-- Dialog Buttons -->
+    <string name="dialog_yes">Yes</string>
```
Comment (teamagam1) [inline app/src/main/res/values/strings.xml:139]: please give more informative names. which dialog do you refer to with dialog_yes? !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java [hunk 2]
```diff
@@ -124,9 +128,44 @@ public void run() {
...
-    private void CalculateCurrentLocation() {
+    private void checkIfGpsEnabled() {
+        final LocationManager manager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+
+        if (!manager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+            buildAlertMessageNoGps();
+        }
+    }
+
+    private void buildAlertMessageNoGps() {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: We would like to clean MainActivity as possible.
please extract this method to other class. maybe create new directory for dialogs. I guess that we will have more from them.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java [hunk 3]
```diff
@@ -124,9 +128,44 @@ public void run() {
...
         //WakefulIntentService.sendWakefulWork(this, GGService.actionGetTipsIntent(this));
 
+        // Check if the GPS is enabled,
+        // and ask the user if he wants to enable it
+        checkIfGpsEnabled();
     }
 
-    private void CalculateCurrentLocation() {
+    private void checkIfGpsEnabled() {
+        final LocationManager manager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: consider moving this method into LocationFetcher. this is where we use locationManager. maybe you can add there flag that will be updated by the LocationManager's receiver.

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (teamagam3) [inline app/src/main/res/values/strings.xml:139]: They're generic string for dialogs

Comment (teamagam3) [inline app/src/main/res/values/strings.xml:139]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 3 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 2 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/TurnOnGpsDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import android.content.Intent;
+
+import com.teamagam.gimelgimel.R;
+
+/**
+ * Represents an alert dialog that shows when the GPS is off
+ */
+public class TurnOnGpsDialogFragment extends AlertDialog {
+
+    private Context mContext;
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/TurnOnGpsDialogFragment.java:15]: you can safely remove mContext.

**Pull request merged**